lib/classification_lifecycle.rb

```diff
@@ -1,82 +1,106 @@
 class ClassificationLifecycle
   class ClassificationNotPersisted < StandardError; end
 
-  attr_reader :classification
-
-  def initialize(classification)
-    @classification = classification
-  end
-
-  def queue(action)
+  def self.queue(classification, action)
     unless classification.persisted?
       message = "Background process called before persisting the classification."
       raise ClassificationNotPersisted.new(message)
     end
     ClassificationWorker.perform_async(classification.id, action.to_s)
   end
 
-  def transact!(&block)
+  def self.perform(classification, action)
```

Comment:
I liked the duality of `self.queue` + `self.perform`, but I don't care too much. Will change.